### PR TITLE
Support edge-to-edge for the filters section

### PR DESCRIPTION
**Pull request by geekygecko**
## Description

This change focuses on converting all the screens in the filters tab to edge-to-edge for the Android 15 project. Please ignore the other tabs for now. I fixed some pages in other tabs, but I will focus on these sections later.

There will be multiple pull requests required to support Android 15 edge-to-edge but I'm going to break them down into smaller PRs. I will merge them all into a `task/android-15` branch, and after all the work is completed, raise a PR to merge into `main`.

## Testing Instructions

Please try tapping around in the filters tab to find issues with the edge-to-edge changes. Areas I have discovered issues with are:
- Older versions of Android versions
- Themes
- Content under the system bars 
- End of pages / lists under the bottom navigation or mini player

Here are some test steps I have followed in case you want to try them too:

1. Open the filters tab
2. ✅ Verify the next pages aren't under the status bar and you can get to the bottom of the page, even with the mini player open.
3. Tap on a filter
4. Tap on the three dots -> Filter settings
5. Go back and create a filter
6. Try the different filter options such as "Episode Status" and "Release Date"

## Screenshots

Android 15, SDK 35

| Before | After |
| --- | --- |
| filters-before-list | filters-after-list |
| filters-before-list-dark | filters-after-list-dark |
| filters-before-list-rose | filters-after-list-rose |
| filters-before-list-classic | filters-after-list-classic |
| filters-before-list-radiation | filters-after-list-radiation |
| filters-before-empty | filters-after-empty |
| filters-before-settings | filters-after-settings |
| filters-before-create | filters-after-create |
| filters-before-choose-podcasts | filters-after-choose-podcasts |
| filters-before-episode-status | filters-after-episode-status |
| filters-before-release-date | filters-after-release-date |
| filters-before-episode-duration | filters-after-episode-duration |
| filters-before-download | filters-after-download |
| filters-before-media-type | filters-after-media-type |
| filters-before-list-landscape | filters-after-list-landscape |

Android 11, SDK 30
| Before | After |
| --- | --- |
| filters-before | filters-after |
| filters-create-before | filters-create-after |
| filters-episode-status-before | filters-episode-status-after |
| filters-dark-before | filters-dark-after |
| filters-create-dark-before | filters-create-dark-after |
| filters-episode-status-dark-before | filters-episode-status-dark-after |
| filters-rose-before | filters-rose-after |
| filters-create-rose-before | filters-create-rose-after |
| filters-episode-status-rose-before | filters-episode-status-rose-after |

## Checklist
- [x] If this is a user-facing change, I have added an entry in CHANGELOG.md
- [x] Ensure the linter passes (`./gradlew spotlessApply` to automatically apply formatting/linting)
- [x] I have considered whether it makes sense to add tests for my changes
- [x] All strings that need to be localized are in `modules/services/localization/src/main/res/values/strings.xml`
- [x] Any jetpack compose components I added or changed are covered by compose previews
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.
 
#### I have tested any UI changes...

- [x] with different themes
- [x] with a landscape orientation
- [x] with the device set to have a large display and font size
- [x] for accessibility with TalkBack

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 500 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: app/src/main/java/au/com/shiftyjelly/pocketcasts/ui/MainActivity.kt [hunk 1]
```diff
@@ -994,27 +987,19 @@ class MainActivity :
...
     private fun updateSnackbarPosition(bottomPadding: Int) {
         binding.snackbarFragment.updateLayoutParams<ViewGroup.MarginLayoutParams> {
-            bottomMargin = bottomPadding
-        }
-    }
-
-    private fun setupSnackbarPosition() {
-        // we manually position the snackbar to account for the mini player and bottom navigation, consume the insets so extra padding isn't added to the snackbar
-        ViewCompat.setOnApplyWindowInsetsListener(binding.snackbarFragment) { _, _ ->
-            WindowInsetsCompat.CONSUMED
```
Comment (geekygecko) [inline app/src/main/java/au/com/shiftyjelly/pocketcasts/ui/MainActivity.kt:1010]: This caused me an issue for a while. I learned it isn't good to consume the insets as nothing else can use them. The filters create page didn't get any insets until I removed this.

File: app/src/main/java/au/com/shiftyjelly/pocketcasts/ui/MainActivity.kt [hunk 2]
```diff
@@ -694,16 +692,11 @@ class MainActivity :
         }
     }
 
-    private fun updateNavAndStatusColors(playerOpen: Boolean, playingPodcast: Podcast?) {
+    private fun updateNavAndStatusColors(playerOpen: Boolean) {
```
Comment (geekygecko) [inline app/src/main/java/au/com/shiftyjelly/pocketcasts/ui/MainActivity.kt:695]: There is no need to pass the podcast as we don't colour the navigation bar as the fragments will now do this.

File: modules/features/discover/src/main/res/layout/fragment_discover.xml [hunk 3]
```diff
@@ -1,45 +1,55 @@
 <?xml version="1.0" encoding="utf-8"?>
-<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (geekygecko) [inline modules/features/discover/src/main/res/layout/fragment_discover.xml:2]: I'm not sure how the Discover page change got in here. I will focus on this section in the next PR.

File: modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/playback/UpNextQueueImpl.kt [hunk 4]
```diff
@@ -206,7 +206,7 @@ class UpNextQueueImpl @Inject constructor(
         val mutableList = episodes.toMutableList()
         if (isEmpty) {
             val firstEpisode = mutableList.first()
-            playNextNow(firstEpisode, downloadManager, null)
```
Comment (geekygecko) [inline modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/playback/UpNextQueueImpl.kt:209]: There isn't a `playNextNow` after the methods were renamed to have a suffix `Blocking`, I think this is an issue I introduced into the `task/android-15` branch.

File: app/src/main/java/au/com/shiftyjelly/pocketcasts/ui/MainActivity.kt [hunk 5]
```diff
@@ -994,27 +987,19 @@ class MainActivity :
...
         binding.snackbarFragment.updateLayoutParams<ViewGroup.MarginLayoutParams> {
-            bottomMargin = bottomPadding
-        }
-    }
-
-    private fun setupSnackbarPosition() {
-        // we manually position the snackbar to account for the mini player and bottom navigation, consume the insets so extra padding isn't added to the snackbar
-        ViewCompat.setOnApplyWindowInsetsListener(binding.snackbarFragment) { _, _ ->
-            WindowInsetsCompat.CONSUMED
+            bottomMargin = bottomPadding + 16.dpToPx(binding.snackbarFragment.context)
```
Comment (MiSikora) [inline app/src/main/java/au/com/shiftyjelly/pocketcasts/ui/MainActivity.kt:996]: Is `16` just a simple space value or does it come from somewhere?

**Review by MiSikora — APPROVED** [review]
I didn't find any issues. Nice job!

Probably unrelated to these changes but on SDK 28 at one point the bottom navigation bar was under the system navigation bar. It was barely visible. Once I tapped on one of the icons it appeared fully.

File: app/src/main/java/au/com/shiftyjelly/pocketcasts/ui/MainActivity.kt (hunk 5 above)
Comment (geekygecko) [reply app/src/main/java/au/com/shiftyjelly/pocketcasts/ui/MainActivity.kt:996]: It's just to add some padding to the snack bar, but I'm currently reworking its position as it isn't quite right. In landscape, it's slightly behind the mini player.

Comment (geekygecko) [reply app/src/main/java/au/com/shiftyjelly/pocketcasts/ui/MainActivity.kt]: |  |  |  |
| --- | --- | --- |
| Screenshot_20241129_221236 | Screenshot_20241129_221225 | Screenshot_20241129_221745 |

I've improved the positioning and removed the 16.dp padding.

**Pull request merged**